### PR TITLE
Sync GitHub actions, other than release actions

### PR DESCRIPTION
**Pull request by trask**
See also https://github.com/open-telemetry/opentelemetry-java-contrib/pull/274 and https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/5695.

Also several updates to the github actions doc.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -74,18 +75,16 @@ jobs:
...
     uses: ./.github/workflows/reusable-misspell-check.yml
 
   publish-snapshots:
-    name: Publish snapshots
-    if: ${{ github.event_name == 'push' && github.repository == 'open-telemetry/opentelemetry-java' }}
-    # snapshot publishing is not dependent on markdown-link-check success, because links to external
-    # urls can break at any time
+    # intentionally not blocking snapshot publishing on markdown-link-check or misspell-check
     needs: build
+    if: ${{ github.event_name == 'push' && github.repository == 'open-telemetry/opentelemetry-java' }}
```
Comment (anuraaga) [inline .github/workflows/build.yml:87]:
```suggestion
    if: ${{ github.repository == 'open-telemetry/opentelemetry-java' }}
```

Doesn't seem to be a reason to not publish snapshots on workflow dispatch

File: .github/workflows/codeql-daily.yml [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
+name: CodeQL (daily)
 
 on:
-  workflow_dispatch:
   schedule:
     - cron: '30 1 * * *'
+  workflow_dispatch:
 
 jobs:
   CodeQL-Build:
```
Comment (anuraaga) [inline .github/workflows/codeql-daily.yml:9]:
```suggestion
  analyze:
```

I think

File: docs/common-github-actions-practices.md [hunk 3]
```diff
@@ -336,40 +378,322 @@ This is what we use in the OpenTelemetry Java repositories:
...
+
+For OpenTelemetry Java repositories, the version in the `main` branch always ends with `-SNAPSHOT`,
+so preparing the release branch involves
+
+* removing `-SNAPSHOT` from the version on the release branch
+  (e.g. updating the version from `1.2.0-SNAPSHOT` to `1.2.0`)
+* bumping the version to the next `-SNAPSHOT` on main
+  (e.g. updating the version from `1.2.0-SNAPSHOT` to `1.3.0-SNAPSHOT`)
+
+```
```
Comment (mateuszrzeszutek) [inline docs/common-github-actions-practices.md:391]: Nit: can we specify the language of the code samples in this file? Makes them a bit easier to read in GH
```suggestion
```yaml
```

**Review by mateuszrzeszutek — APPROVED** [review]

File: docs/common-github-actions-practices.md [hunk 4]
```diff
@@ -709,8 +716,10 @@ Use `.yml` extension instead of `.yaml`.
...
 * `.github/workflows/codeql-daily.yml`
 
-TODO other common names?
+## Workflow YAML style guide
 
-## YAML style guide
+Workflow names - [Sentence case](https://en.wikipedia.org/wiki/Letter_case#Sentence_case)
 
-TODO
+Job names - [kebab-case(https://en.wikipedia.org/wiki/Letter_case#Kebab_case)
```
Comment (mateuszrzeszutek) [inline docs/common-github-actions-practices.md:723]:
```suggestion
Job names - [kebab-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case)
```

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

**Pull request merged**